src/EnergyPlus/Coils/CoilCoolingDX.cc

```diff
@@ -575,15 +575,15 @@ void CoilCoolingDX::oneTimeInit(EnergyPlusData &state)
 int CoilCoolingDX::getNumModes()
 {
     int numModes = 1;
-    if (this->performance.hasAlternateMode) {
+    if (this->performance.coilMode != HVAC::CoilMode::Normal) {
         numModes++;
     }
```

Comment:
This looks sound.

    if (!input_data.alternate_operating_mode_name.empty() && input_data.alternate_operating_mode2_name.empty()) {
        this->hasAlternateMode = DataHVACGlobals::coilEnhancedMode;

    if (!input_data.alternate_operating_mode2_name.empty() && !input_data.alternate_operating_mode_name.empty()) {
        this->hasAlternateMode = DataHVACGlobals::coilSubcoolReheatMode;